### PR TITLE
Issue 211 - points 1 and 3

### PR DESCRIPTION
**Pull request by adarshp**
This PR addresses points 1 and 3 of #211 .

### CODE REVIEW

File: tests/data/crop_yield.json [hunk 1]
```diff
@@ -433,13 +450,31 @@
...
         {
           "name": "crop_yield__loop_plate__day_0",
-          "inputs": [
-            "consistency",
-            "max_rain",
-            "absorption",
-            "rain",
-            "total_rain",
-            "yield_est"
+          "input": [
```
Comment (pauldhein) [inline tests/data/crop_yield.json]: I wasn't looking for index information in the input list. I am sorry if I didn't make that clear. The objects hear should just have the name and domain information.

File: tests/data/PETPT.json [hunk 2]
```diff
@@ -207,17 +207,17 @@
...
         {
-          "name": "eo_2",
+          "name": "eo_3",
           "type": "variable"
         }
       ]
     },
     {
-      "name": "petpt__assign__eo_5",
+      "name": "petpt__assign__eo_6",
```
Comment (pauldhein) [inline tests/data/PETPT.json]: It seems like we have introduced an indexing bug that is indexing all variables too quickly. We should only have `eo_0` through `eo_5` represented here, but instead we are seeing `eo_6`.

File: delphi/translators/for2py/genPGM.py [hunk 3]
```diff
@@ -222,13 +222,29 @@ def genPgm(self, node, state, fnNames, call_source):
...
                 "body": loopBody,
             }
 
-            loopCall = {"name": loopName, "inputs": variables, "output": {}}
+            loopCall = {
+                "name": loopName,
+                "input": [
+                    {
+                      "name": variable,
+                      "index": -1,
```
Comment (pauldhein) [inline delphi/translators/for2py/genPGM.py]: This line can probably just be changed to `"domain": state.varTypes[variable]` for now

File: tests/data/PETPT.json (hunk 2 above)
Comment (adarshp) [reply tests/data/PETPT.json]: @pratikbhd Looks too hard for me to fix - I'll leave it for you and close this PR :)

**Comment by pauldhein** [comment]
@adarshp as noted in #211, I messed up in reviewing your changes (I got confused by the git diff view instead of the actual file view). This PR should be re-opened and merged with master (after solving the off-by-1 indexing bug that was introduced to PETPT), but the input list for the loop plate call in the body of the crop yield container is formed correctly.

**Comment by pratikbhd** [comment]
> @adarshp as noted in #211, I messed up in reviewing your changes (I got confused by the git diff view instead of the actual file view). This PR should be re-opened and merged with master (after solving the off-by-1 indexing bug that was introduced to PETPT), but the input list for the loop plate call in the body of the crop yield container is formed correctly.

Would it be alright if I implemented this change in the branch that I am working on right now regarding the indexing issue for #211? That would make the merges more streamlined.

**Comment by pauldhein** [comment]
@pratikbhd that sounds like a good idea to me. BTW, any idea why the indexing error started to occur in `PETPT`? It doesn't seem to me like @adarshp changed anything, but yet the `PETPT` `EO` indices seem to start at 1 instead of 0.

**Pull request closed without merging**